andela/ah-lobos-backend
# #166789869 Email verification after signup

#### What does this PR do?
This PR does the following:
- test the verification email method
- add send verification method as a helper
- add verification method as a controller

#### Description of Task to be completed?
 - The email verification is found when the user signs up and he/she must get a confirmation email
 - if the user clicks the link he/she must get a response that shows he/she is verified

#### How should this be manually tested?
> Follow the setup instructions laid out in the ```README.md``` file. 

> run ``` npm run start ```

> access the user sign-up route via `/api/users`

> check the user email and click the link

or signup using Heroku link: <https://ah-lobos-backend-swagger.herokuapp.com>

> boom, you are now verified.

#### Any background context you want to provide?

- use Nodemailer package for email verification 

#### What are the relevant pivotal tracker stories?

[#166789869](https://www.pivotaltracker.com/story/show/166789869)

#### Screenshots (if appropriate)
> swagger images for testing the signup
![image](https://user-images.githubusercontent.com/31614647/60747054-f8c90800-9f82-11e9-9d5f-ab035a357ee3.png)
![image](https://user-images.githubusercontent.com/31614647/60747065-0ed6c880-9f83-11e9-83d1-d0a7d3a6308c.png)
> Message was sent to the user email image
![image](https://user-images.githubusercontent.com/31614647/60748450-629adf00-9f8e-11e9-9a31-f0a54c19450a.png)
 
#### Questions: 
N/A
